etna/metrics/base.py

```diff
@@ -339,4 +362,93 @@ def __call__(self, y_true: TSDataset, y_pred: TSDataset) -> Union[float, Dict[st
         return metrics
 
 
-__all__ = ["Metric", "MetricAggregationMode"]
+class MetricWithMissingHandling(Metric):
```

Comment:
We should also probably do smth with NaNs returned from metric functions:
- We should cast them into `None` inside `__call__`
- We should cast `NaNs` into `None` inside the functional metrics